### ISSUE TITLE
Cannot deploy container app using images from azure container registry in different subscription

### ISSUE THREAD

**Issue by mick-anderson24** [microsoft/azure-container-apps#1156]
> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is a: (mark with an x)
- [X ] bug report -> please search issues before submitting
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)

### Issue description
We have an azure container registry running in a private endpoint on one subscription (Subscription A) and have the azure container apps environment running in a private endpoint on another subscription (subscription B).  We are attempting to create a container app using an image in the Subscription A using a user managed identity to pull the image and deploy it out.  When running the Azure CLI command we are getting this error: The following field(s) are either invalid or missing. Field 'template.containers.ca-product-definition-platform.image' is invalid with details: 'Invalid value: "{{RegisteryUrI}}/root/{{image with version}}": unable to pull image using Managed identity /subscriptions/{{SubscriptionId}}/resourcegroups/{{ResourceGroup}}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{{UserAssignedIdentity}} for registry {{RegistryUri}}'; .  I've looked at issues for this workaround [719](https://github.com/microsoft/azure-container-apps/issues/719) but still have had no luck

### Steps to reproduce
1. Create Azure container registry in subscription A within a private endpoint
2. Create azure container app environment in subscription B within a private endpoint
3. Create a user assigned managed identity in subscription B and grant it ACRPull rights on the container registry
4. Run Azure CLI command: az containerapp create -n $ContainerAppName  -g $ResourceGroupName --image {{ACR Name}}/root/{{image with version}} --environment {{ContainerAppsEnvironment}} --registry-server {{ACR Name}} --registry-identity /subscriptions/{{SubscriptionB ID}}/resourcegroups/{{ResourceGroup Name}}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{{UserAssignedIdentity}} --ingress internal --target-port 8080

**Expected behavior** 
I would expect to be able to connect to my Azure container registry using the user assigned managed identity to pull the image and deploy the container into the container apps environment.  I would also be nice to have the ability to configure this same option via the Azure Portal but that is also not possible when the registry lives in a different subscription than the container apps

**Actual behavior** 
 The following field(s) are either invalid or missing. Field 'template.containers.ca-product-definition-platform.image' is invalid with details: 'Invalid value: "{{RegisteryUrI}}/root/{{image with version}}": unable to pull image using Managed identity /subscriptions/{{SubscriptionId}}/resourcegroups/{{ResourceGroup}}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{{UserAssignedIdentity}} for registry {{RegistryUri}}';

**Screenshots**         

### Additional context
Azure CLI but this also is impossible to do in the portal due to it only allowing the lookup for registries in the same subscription

**Comment by vturecek** [comment]
@mick-anderson24 does your ACR allow ARM audience tokens? https://learn.microsoft.com/en-us/azure/container-apps/managed-identity-image-pull?tabs=azure-cli&pivots=azure-portal#prerequisites

Also, is your Container App Environment in your own VNET?

**Comment by mick-anderson24** [comment]
> @mick-anderson24 does your ACR allow ARM audience tokens? https://learn.microsoft.com/en-us/azure/container-apps/managed-identity-image-pull?tabs=azure-cli&pivots=azure-portal#prerequisites
> 
> Also, is your Container App Environment in your own VNET?

Yes to both of your questions.  The vnets for both subscriptions are peered so they should be able to communicate with each other

**Comment by dhasek00** [comment]
Hi I was just fighting this same issue over the past day. I previously had a working container app that could pull from an ACR in a different subscription, but tried to automate it in Terraform this week and ran into trouble.

I found a difference between my private endpoint created in Terraform vs the one created from the portal (which worked) was missing a DNS Configuration with a Private DNS Zone (PrivateLink). I noticed when I went back to create the endpoint in the portal that one of the steps wanted to make this.

Once I added that to the ACR private endpoint the App was able to pull from the registry again.

**Comment by chinadragon0515** [comment]
@mick-anderson24 seems the UAMI does not have permission to pull the image, do you still have issue? If yes, can you send your container app info and the timestamp of the issue to acasupport(at)microsoft(dot)com
